server/service/setup_experience.go

```diff
@@ -214,3 +214,47 @@ func (svc *Service) DeleteSetupExperienceScript(ctx context.Context, teamID *uin
 
 	return fleet.ErrMissingLicense
 }
+
+// maybeUpdateSetupExperienceStatus attempts to update the status of a setup experience result in
+// the database. If the given result is of a supported type (namely SetupExperienceScriptResult,
+// SetupExperienceSoftwareInstallResult, and SetupExperienceVPPInstallResult), it returns a boolean
+// indicating whether the datastore was updated and an error if one occurred. If the result is not of a
+// supported type, it returns false and an error indicated that the type is not supported.
+// If the skipPending parameter is true, the datastore will only be updated if the given result
+// status is not pending.
+func maybeUpdateSetupExperienceStatus(ctx context.Context, ds fleet.Datastore, result interface{}, requireTerminalStatus bool) (bool, error) {
```

Comment:
Maybe we could create a `SetupExperienceResult` interface that contains the `SetupExperienceStatus`,
instead of taking an `interface{}` here?